Assets/Spawner.cs

```diff
@@ -4,12 +4,70 @@
 
 public class Spawner : MonoBehaviour
 {
+    public GameObject PlatformPrefab;
+    public GameObject FinishPrefab;
     public GameObject CoinPrefab;
-    
+    float jump_length = 4.5f; // Approximate jump length given speed, jump speed, and gravitational constant derived from expiration.
+
     void Start() {
-        for (int i = 0; i < 100; i++) {
-            var position = new Vector3(0, 0, i);
-            Instantiate(CoinPrefab, position, Quaternion.identity);
+        int platform_number = Random.Range(10, 31);
+        float pos_x = 0.0f;
+        float pos_z = 0.0f;
+        float direction = 0.0f; // Alternating 0.0f and 90.0f
+
+        for (int i = 0; i < platform_number; i++) {
+            float length;
+            if (i == 0)
+                length = 12.0f; // First platform is long to prepare for game
+            else
+                length = Random.Range(2.0f, 6.0f);
+            var position = new Vector3( // The origin of block is center of it, hence add the half of size of block
```

Comment:
comments are nice, but if you want them to describe your intent, you should extract pieces of your code into methods with meaningful names.